pihole

```diff
@@ -95,8 +95,7 @@ uninstallFunc() {
 
 versionFunc() {
   shift
-  "${PI_HOLE_SCRIPT_DIR}"/version.sh "$@"
-  exit 0
+  exec "${PI_HOLE_SCRIPT_DIR}"/version.sh "$@"
```

Comment:
Is there a special reason to use `exec` here?

```suggestion
  "${PI_HOLE_SCRIPT_DIR}"/version.sh "$@"
  exit $?
```

Reply:
This is to provide an exit code, so that `return 1` in the functions/`exit 1` in `version.sh` also leads to an `exit 1` of the `pihole` command. `exit $?` is the obvious alternative.

The "benefit" of `exec` is that it replaces the invoking process instead of spawning a child. So there are no two processes running but only one, less overhead and one does not need to pass the exit code from the child to the parent. So if a command is assured to be the last command in a script, `exec` simply always makes sense.

Reply:
Then we should probably change it here too

https://github.com/pi-hole/pi-hole/blob/b5e0f142ccefe9b1b59b94d7e82d3052efbfdcd0/pihole#L73-L75


But this is for another PR